code/game/objects/structures/maintenance.dm

```diff
@@ -105,3 +105,129 @@ at the cost of risking a vicious bite.**/
+		return
+	return ..()
+
+/obj/structure/destructible/cult/pants_altar/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(.)
+		return
+	var/list/altar_options = list(
+		"Change Color" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_recolor"),
+		"Create Artefact" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_create")
+	)
+	var/altar_result = show_radial_menu(user, src, altar_options, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	switch(altar_result)
+		if("Change Color")
+			var/chosen_color = input(usr, "", "Choose Color", pants_color) as color|null
+			if(!isnull(chosen_color) && user.canUseTopic(src, BE_CLOSE))
+				pants_color = chosen_color
+		if("Create Artefact")
+			if(!COOLDOWN_FINISHED(src, use_cooldown_duration))
+				to_chat(usr, "<span class='warning'>[src] is not ready to create something new yet...</span>")
+				return
+			pants_stageone()
+	return TRUE
+
+/obj/structure/destructible/cult/pants_altar/update_icon_state()
+	. = ..()
+	if(!COOLDOWN_FINISHED(src, use_cooldown))
+		icon_state = "altar_off"
+	else
+		icon_state = "altar"
+
+/obj/structure/destructible/cult/pants_altar/update_overlays()
+	. = ..()
+	var/overlayicon
+	switch(status)
+		if(ALTAR_INACTIVE)
+			return
+		if(ALTAR_STAGEONE)
+			overlayicon = "altar_pants1"
+		if(ALTAR_STAGETWO)
+			overlayicon = "altar_pants2"
+		if(ALTAR_STAGETHREE)
+			overlayicon = "altar_pants3"
+	var/mutable_appearance/pants_overlay = mutable_appearance(icon, overlayicon)
+	pants_overlay.appearance_flags = RESET_COLOR
+	pants_overlay.color = pants_color
+	. += pants_overlay
+
+/obj/structure/destructible/cult/pants_altar/proc/pants_stageone()
+	status = ALTAR_STAGEONE
+	update_icon()
+	visible_message("<span class='warning'>[src] starts creating something...</span>")
+	playsound(src, 'sound/magic/pantsaltar.ogg', 60)
+	addtimer(CALLBACK(src, .proc/pants_stagetwo), ALTAR_TIME)
+
+/obj/structure/destructible/cult/pants_altar/proc/pants_stagetwo()
+	status = ALTAR_STAGETWO
+	update_icon()
+	visible_message("<span class='warning'>You start feeling nauseous...</span>")
+	for(var/mob/living/mob in viewers(7, src))
+		mob.blur_eyes(10)
+		mob.add_confusion(10)
+	addtimer(CALLBACK(src, .proc/pants_stagethree), ALTAR_TIME)
+
+/obj/structure/destructible/cult/pants_altar/proc/pants_stagethree()
+	status = ALTAR_STAGETHREE
+	update_icon()
+	visible_message("<span class='warning'>You start feeling horrible...</span>")
+	for(var/mob/living/mob in viewers(7, src))
+		mob.set_timed_status_effect(20 SECONDS, /datum/status_effect/dizziness, only_if_higher = TRUE)
+	addtimer(CALLBACK(src, .proc/pants_create), ALTAR_TIME)
+
+/obj/structure/destructible/cult/pants_altar/proc/pants_create()
+	status = ALTAR_INACTIVE
+	update_icon()
+	visible_message("<span class='warning'>[src] emits a flash of light and creates... pants?</span>")
+	for(var/mob/living/mob in viewers(7, src))
```

Comment:
while you're doing other stuff, might as well improve this var name which is kinda meh